packages/eslint-plugin/src/index.js

```diff
@@ -1,6 +1,10 @@
 // This registers Typescript compiler instance onto node.js.
 // Now it is possible to just require typescript files without any compilation steps in the environment run by node
-require("ts-node").register();
+require("ts-node").register({
+  compilerOptions: {
+    module: "commonjs",
```

Comment:
force ts-node to compile our ESLint plugin's TS files into CommonJS (which ESLint requires)